### PR TITLE
feature: Implementing chat authentication

### PR DESCRIPTION
**Pull request by kimz95**

### CODE REVIEW

File: webserver/src/chat/mod.rs [hunk 1]
```diff
@@ -7,28 +7,47 @@ use std::{
...
+            }
+            }else{
+                log::error!("Invalid token in header!");
+                Err(ErrorBadRequest("Invalid token format. Expected 'Bearer <token>'"))
+            }
+        }
+        else{
+            log::error!("Authorization header is missing");
+            Err(ErrorUnauthorized("Authorization header is missing"))
+        }
```
Comment (dIB59) [inline webserver/src/chat/mod.rs:52]: Return only a single error, just say unauthorized. No reason to state anything else.

File: webserver/src/chat/mod.rs [hunk 2]
```diff
@@ -7,28 +7,47 @@ use std::{
...
-    )
+    if let Some(auth_header) = req.headers().get("Authorization") {
+        if let Some(token) = auth_header.to_str().expect("Invalid UTF-8 Auth header").strip_prefix("Bearer "){
+            match decode_jwt(&token) {
+                Ok(claims) => ws::start(
+                    chat::WsChatSession {
+                            id: 0,
+                            hb: Instant::now(),
+                            room: "main".to_owned(),
+                            name: None,
```
Comment (dIB59) [inline webserver/src/chat/mod.rs:33]: Name should not be none, it should be the user name.

**Review by dIB59 — CHANGES_REQUESTED** [review]
When please use match statements, look at the rust way of doing things

**Comment by kimz95** [comment]
> When please use match statements, look at the rust way of doing things

Should I use nested match statements in cases like these or is better to use 'unwrap_or' for error handling? because I seem to find contradicting things online about best practices

**Review by dIB59 — APPROVED** [review]
LGTM

**Pull request closed without merging**